src/Project.cpp

```diff
@@ -178,11 +194,24 @@ void Project::load(const nlohmann::json& j) {
 
     // parse worlds
     auto external_levels = j["externalLevels"].get<bool>();
+
+    if (external_levels && was_loaded_from_memory) {
+        ldtk_error("External levels are not supported when loading from memory.");
+    }
+
     if (j["worlds"].empty()) {
-        m_worlds.emplace_back(j, this, external_levels);
+        if (file_loader.has_value()) {
+            m_worlds.emplace_back(j, this, external_levels, file_loader.value());
+        } else {
+            m_worlds.emplace_back(j, this);
+        }
```

Comment:
Here, not using optional would greatly simplify this code block to what it was before by directly passing `file_loader`.